pkg/volume/csi/csi_plugin.go

```diff
@@ -59,6 +58,8 @@ const (
 	CsiResyncPeriod = time.Minute
 )
 
+var csiTimeout = 2 * time.Minute
```

Comment:
Hello. This code change does not do anything new.  What is the motivation for this ?

Reply:
To make it variable,so we can make it faster when running unit tests.

Reply:
Instead of making `csiTimeout` a global var (which could introduce wanted race issues), I would suggest making it a member of the `csiPlugin` struct here:

https://github.com/kubernetes/kubernetes/blob/4e8a7f4a4b0cf7705d8e3cea9670bed85be65d20/pkg/volume/csi/csi_plugin.go#L62

That way it can be better managed.  

Another issue to consider with the timeout is that each CSI driver will have different backends which will have different timeout requirements. So another interesting approach is to (maybe) read the timeout value from the storage class parameters ?

Reply:
> Instead of making `csiTimeout` a global var (which could introduce wanted race issues), I would suggest making it a member of the `csiPlugin` struct here:
> 
> https://github.com/kubernetes/kubernetes/blob/4e8a7f4a4b0cf7705d8e3cea9670bed85be65d20/pkg/volume/csi/csi_plugin.go#L62
> 
> That way it can be better managed.
> 
> Another issue to consider with the timeout is that each CSI driver will have different backends which will have different timeout requirements. So another interesting approach is to (maybe) read the timeout value from the storage class parameters ?

Seems like a good idea, but if the value is too small, e.g. 1ms, this may increase the load of the system. 
So I would like just add to `csiPlugin`

Reply:
Please add variable `csiTimeout` to the `csiPlugin` struct.

Reply:
Maybe can not only add `csiTimeout` to `csiPlugin`. Many other places also use it but they do not belong to `csiPlugin`. e.g.

https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/csi/csi_attacher.go#L124
https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/csi/csi_metrics.go#L52


Do you think we also need add it to other `struct` or just change `const` to `variable`